python/src/plugins/jupiter/goat_plugins/jupiter/service.py

```diff
@@ -0,0 +1,114 @@
+import base64
+from typing import Dict, Any
+
+import aiohttp
+from goat.decorators.tool import Tool
+from solders.transaction import VersionedTransaction
+from .parameters import GetQuoteParameters, QuoteResponse, SwapParameters
+from goat_wallets.solana import SolanaWalletClient
+
+
+class JupiterService:
+    def __init__(self, api_key: str):
+        self.api_key = api_key
+        self.base_url = "https://quote-api.jup.ag/v6"
+        self._timeout = aiohttp.ClientTimeout(total=10)  # 10 second timeout
+        self._session_kwargs = {"timeout": self._timeout}
+
+    @Tool({
+        "description": "Get a quote for a swap on the Jupiter DEX",
+        "parameters_schema": GetQuoteParameters
+    })
+    async def get_quote(self, wallet_client: SolanaWalletClient, parameters: dict) -> QuoteResponse:
+        """Get a quote for swapping tokens using Jupiter."""
+        try:
+            params = GetQuoteParameters.parse_obj(parameters)
```

Comment:
The method "parse_obj" in class "BaseModel" is deprecated
  The `parse_obj` method is deprecated; use `model_validate` instead